runtime/resolvers/resolvers_test.go

```diff
@@ -0,0 +1,158 @@
+package resolvers
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"flag"
+	"maps"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"testing"
+
+	"github.com/rilldata/rill/runtime"
+	"github.com/rilldata/rill/runtime/testruntime"
+	"github.com/stretchr/testify/require"
+	"gopkg.in/yaml.v3"
+)
+
+type Project struct {
+	Sources    map[string]yaml.Node
+	Models     map[string]yaml.Node
+	Dashboards map[string]yaml.Node
+	APIs       map[string]yaml.Node
+}
+type Resolvers struct {
+	Project    Project
+	Connectors map[string]*testruntime.InstanceOptionsForResolvers
+	Tests      map[string]*Test
+}
+
+type Test struct {
+	Resolver string
+	Options  runtime.ResolveOptions
+	Result   []map[string]any
+}
```

Comment:
I strongly believe it would simplify the code and readability here to combine this into a single struct with anonymous sub-structs to represent the YAML structure – and avoid using internal types.

This is also what we do for the resource YAML parsing, and it serves us well. Look at the `AlertYAML` type here as an example: https://github.com/rilldata/rill/blob/d66a28ff94b55273d102cc2112faf8d05465d254/runtime/compilers/rillv1/parse_alert.go#L18

Reply:
OK, but I'll need to copy all the data to ResolveOptions anyway here to call a method:
```
					res, err := rt.Resolve(context.Background(), &ropts)
```

Reply:
`SecurityClaims` have a `oneof` field, now it makes sense.

Reply:
Done.